experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c

```diff
@@ -339,6 +339,7 @@ static int ws_eink_init_display(struct ws_eink_fb_par *par)
 static int ws_eink_update_display(struct ws_eink_fb_par *par)
 {
 	int ret = 0;
```

Comment:
I think you have added a lot more code that you needed to.
```
u8 *vmem = par->info->screen_base;
u8 *ssbuf = par->ssbuf;
const u8 *lut;
size_t lut_size;
static int update_count = 0;
if (++update_count == 10) {
	update_count = 0;
	lut = lut_full_update;
	lut_size = ARRAY_SIZE(lut_full_update);
} else {
	lut = lut_partial_update;
	lut_size = ARRAY_SIZE(lut_partial_update);
}
ret = int_lut(par, lut, lut_size);
if (ret)
	return ret;
```
Rest of the function remains the same.